util/types/time.go

```diff
@@ -1202,16 +1202,14 @@ func checkTimestampType(t TimeInternal) error {
 		return nil
 	}
 
-	// TODO: Consider time_zone variable.
-	t1, err := t.GoTime(gotime.Local)
-	if err != nil {
-		log.Infof("checkTimestampType failed, t=%v", t)
-		return errors.Trace(err)
+	if compareTime(t, maxTimestamp) > 0 || compareTime(t, minTimestamp) < 0 {
+		return errors.Trace(ErrInvalidTimeFormat)
 	}
 
-	if t1.After(maxTimestamp) || t1.Before(minTimestamp) {
-		return ErrInvalidTimeFormat
+	if _, err := t.GoTime(gotime.Local); err != nil {
```

Comment:
Yes.
This is just a test, the converted value is never used.
If `t` can be converted to go time, it should be a valid time format.